src/main/java/com/google/ehub/data/EntertainmentItemDatastore.java

```diff
@@ -12,54 +13,97 @@
-  public static final String IMAGE_URL_PROPERTY_KEY = "imageURL";
+  private final String ENTERTAINMENT_ITEM_KIND = "entertainmentItem";
+  private final String UNIQUE_ID_PROPERTY_KEY = "uniqueID";
+  private final String TITLE_PROPERTY_KEY = "title";
+  private final String LOWERCASE_TITLE_PROPERTY_KEY = "lowercaseTitle";
+  private final String DESCRIPTION_PROPERTY_KEY = "description";
+  private final String IMAGE_URL_PROPERTY_KEY = "imageURL";
+
+  private final DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+
+  private static EntertainmentItemDatastore instance;
+
+  private EntertainmentItemDatastore() {}
+
+  /**
+   * Gives access to the single instance of the class, and creates this instance if it was not
+   * initialized previously.
+   *
+   * @return single instance of the class
+   */
+  public static EntertainmentItemDatastore getInstance() {
+    if (instance == null) {
+      instance = new EntertainmentItemDatastore();
+    }
 
-  private static DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+    return instance;
+  }
 
   /**
    * Adds an EntertainmentItem to Datastore.
    *
-   * @param title title of the entertainment item
-   * @param description description of the entertainment item
-   * @param imageURL URL to the image of the entertainment item stored in Blobstore
+   * @param entertainmentItem entertainment item to add into Datastore
    */
-  public static void addEntertainmentItemToDatastore(
-      String title, String description, String imageURL) {
+  public void addItemToDatastore(EntertainmentItem entertainmentItem) {
     Entity entertainmentItemEntity = new Entity(ENTERTAINMENT_ITEM_KIND);
 
-    entertainmentItemEntity.setProperty(TITLE_PROPERTY_KEY, title);
-    entertainmentItemEntity.setProperty(LOWERCASE_TITLE_PROPERTY_KEY, title.toLowerCase());
-    entertainmentItemEntity.setProperty(DESCRIPTION_PROPERTY_KEY, description);
-    entertainmentItemEntity.setProperty(IMAGE_URL_PROPERTY_KEY, imageURL);
+    entertainmentItemEntity.setProperty(TITLE_PROPERTY_KEY, entertainmentItem.getTitle());
+    entertainmentItemEntity.setProperty(
+        LOWERCASE_TITLE_PROPERTY_KEY, entertainmentItem.getTitle().toLowerCase());
+    entertainmentItemEntity.setProperty(
+        DESCRIPTION_PROPERTY_KEY, entertainmentItem.getDescription());
+    entertainmentItemEntity.setProperty(IMAGE_URL_PROPERTY_KEY, entertainmentItem.getImageURL());
 
     datastoreService.put(entertainmentItemEntity);
   }
 
+  /**
+   * Finds a single entertainment item based on unique id.
+   *
+   * @param uniqueID id used to identify the EntertainmentItem Entity in the Datastore
+   * @return the EntertainmentItem found in Datastore wrapped in an {@link Optional}, the
+   * optional object will be empty if the EntertainmentItem Entity was not found
+   */
+  public Optional<EntertainmentItem> querySingleItem(long uniqueID) {
```

Comment:
nit: "Single" is implied by your return object. Just queryItem seems fine

Reply:
I removed the unnecessary term "Single" in last commit.